cmd/juju/caas/add.go

```diff
@@ -363,10 +388,45 @@ var noRecommendedStorageErrMsg = `
 
 // Run is defined on the Command interface.
 func (c *AddCAASCommand) Run(ctx *cmd.Context) (err error) {
+	if c.hostCloudRegion != "" {
```

Comment:
See previous comment about retaining this attribute name.
Can we just print the deprecation warning when the flags are parsed? Why here as well?